src/Rings/PBWAlgebraQuo.jl

```diff
@@ -57,46 +109,47 @@ end
 ####
 
 function ngens(Q::PBWAlgQuo)
-  return ngens(base_ring(Q))
+  return ngens(base_ring(Q))  # EQUIV  ngens(Q.sring)  ???
 end
 
 function gens(Q::PBWAlgQuo)
-  return elem_type(Q)[PBWAlgQuoElem(Q, x) for x in gens(base_ring(Q))]
+  return elem_type(Q)[PBWAlgQuoElem(Q, PBWAlgElem(Q.I.basering,x)) for x in gens(Q.sring)]
```

Comment:
Meta note: if at all possible, we should move away from directly accessing struct members like `Q.sring` or `Q.I.basering`; instead accessors should be used (and added if necessary). That way, it becomes easier to change internals of structs in the future.